osgp/shared/osgp-ws-smartmetering/src/main/resources/bindings/common.xjb

```diff
@@ -24,6 +24,34 @@
     <jxb:bindings node="//xs:complexType[@name='ProfileEntries']//xs:element[@name='ProfileEntry']">
       <jxb:property name="profileEntries" />
     </jxb:bindings>
+
+    <jxb:bindings node="//xs:simpleType[@name='AlarmType']/xs:restriction/xs:enumeration[@value='PHASE_OUTAGE_DETECTED_L1']">
+      <jxb:typesafeEnumMember name="PHASE_OUTAGE_DETECTED_L1"/>
+    </jxb:bindings>
+    <jxb:bindings node="//xs:simpleType[@name='AlarmType']/xs:restriction/xs:enumeration[@value='PHASE_OUTAGE_DETECTED_L2']">
+      <jxb:typesafeEnumMember name="PHASE_OUTAGE_DETECTED_L2"/>
+    </jxb:bindings>
+    <jxb:bindings node="//xs:simpleType[@name='AlarmType']/xs:restriction/xs:enumeration[@value='PHASE_OUTAGE_DETECTED_L3']">
+      <jxb:typesafeEnumMember name="PHASE_OUTAGE_DETECTED_L3"/>
+    </jxb:bindings>
+    <jxb:bindings node="//xs:simpleType[@name='AlarmType']/xs:restriction/xs:enumeration[@value='VOLTAGE_SAG_IN_PHASE_DETECTED_L1']">
+      <jxb:typesafeEnumMember name="VOLTAGE_SAG_IN_PHASE_DETECTED_L1"/>
+    </jxb:bindings>
+    <jxb:bindings node="//xs:simpleType[@name='AlarmType']/xs:restriction/xs:enumeration[@value='VOLTAGE_SAG_IN_PHASE_DETECTED_L2']">
+      <jxb:typesafeEnumMember name="VOLTAGE_SAG_IN_PHASE_DETECTED_L2"/>
+    </jxb:bindings>
+    <jxb:bindings node="//xs:simpleType[@name='AlarmType']/xs:restriction/xs:enumeration[@value='VOLTAGE_SAG_IN_PHASE_DETECTED_L3']">
+      <jxb:typesafeEnumMember name="VOLTAGE_SAG_IN_PHASE_DETECTED_L3"/>
+    </jxb:bindings>
+    <jxb:bindings node="//xs:simpleType[@name='AlarmType']/xs:restriction/xs:enumeration[@value='VOLTAGE_SWELL_IN_PHASE_DETECTED_L1']">
+      <jxb:typesafeEnumMember name="VOLTAGE_SWELL_IN_PHASE_DETECTED_L1"/>
+    </jxb:bindings>
+    <jxb:bindings node="//xs:simpleType[@name='AlarmType']/xs:restriction/xs:enumeration[@value='VOLTAGE_SWELL_IN_PHASE_DETECTED_L2']">
+      <jxb:typesafeEnumMember name="VOLTAGE_SWELL_IN_PHASE_DETECTED_L2"/>
+    </jxb:bindings>
+    <jxb:bindings node="//xs:simpleType[@name='AlarmType']/xs:restriction/xs:enumeration[@value='VOLTAGE_SWELL_IN_PHASE_DETECTED_L3']">
+      <jxb:typesafeEnumMember name="VOLTAGE_SWELL_IN_PHASE_DETECTED_L3"/>
+    </jxb:bindings>
```

Comment:
I'm wondering if there's a specific reason for having these bindings here, as bindings for the other alarm types are not present?

Reply:
Yes, because if enum values that end with a number as last character will be generated different.

VOLTAGE_SWELL_IN_PHASE_DETECTED_L3 -> VOLTAGE_SWELL_IN_PHASE_DETECTED_L_3
So all values with a number as last character are added here, so this will result in:
VOLTAGE_SWELL_IN_PHASE_DETECTED_L3 -> VOLTAGE_SWELL_IN_PHASE_DETECTED_L3